### PR TITLE
add autofill command

### PR DESCRIPTION
**Pull request by Sneezry**
https://github.com/Authenticator-Extension/Authenticator/discussions/829

### CODE REVIEW

**Comment by mymindstorm** [comment]
So, the reason we didn't implement this before was because we don't have secure website matching like the password managers do.  We can't be sure that a website will not intentionally trick our auto fill matching and pretend to be a legitimate website. Do you think that is a big enough issue to not implement this feature?

**Comment by Sneezry** [comment]
> So, the reason we didn't implement this before was because we don't have secure website matching like the password managers do. We can't be sure that a website will not intentionally trick our auto fill matching and pretend to be a legitimate website. Do you think that is a big enough issue to not implement this feature?

The current design is if there is only one matched account, autofill command will fill the code, or nothing happens. I don't think it's a big problem because in most cases, there's only one matched account. Smart filter works well, and secure website matching is not so important. FYI, this PR is for this feedback https://github.com/Authenticator-Extension/Authenticator/discussions/829

**Comment by philippthiele** [comment]
I am a little confused, my authenticator extension already has such an option, I enabled it to use autofill, but it doesn't do anything. Now I see here it is not really merged yet. What does the current autofill option do?

EDIT:
Oh I see now, I had to restart chrome and now when I click the code it autofills. Sorry for the too quick question. But I will be very interested to see this feature here merged, I think it is a good addition, since I already thought this is what it is doing with the current autofill.

File: src/background.ts [hunk 1]
```diff
@@ -560,6 +562,59 @@ chrome.commands.onCommand.addListener(async (command: string) => {
...
+          if (!tab || !tab.id) {
+            return;
+          }
+          contentTab = tab;
+
+          const siteName = await getSiteName();
+          const entries = await EntryStorage.get();
+          const matchedEntries = getMatchedEntries(siteName, entries);
+
+          if (matchedEntries && matchedEntries.length === 1) {
```
Comment (mymindstorm) [inline src/background.ts:595]: matchedEntries.length <= 1

File: src/background.ts [hunk 2]
```diff
@@ -560,6 +562,59 @@ chrome.commands.onCommand.addListener(async (command: string) => {
...
+              }
+            );
+          } catch (error) {
+            console.error(error);
+            return reject(error);
+          }
+        }
+      );
+
+      if (cachedPassphrase === "") {
```
Comment (mymindstorm) [inline src/background.ts]: This can be removed since you account for unencrypted entries.

File: src/background.ts (hunk 1 above)
Comment (Sneezry) [reply src/background.ts:595]: if `matchedEntries.length == 0`, `const entry = matchedEntries[0]` will be `undefined`. We should auto fill code when there is only one matched entry

Comment (mymindstorm) [reply src/background.ts:595]: Sorry, incorrect logic on my end. I meant `matchedEntries.length >= 1`. It won't past a code if there is more than one match.

Comment (Sneezry) [reply src/background.ts:595]: If there is more than one matched entry I think we should always ask the user to choose the correct one, but not the first one. What do you think?

Comment (mymindstorm) [reply src/background.ts:595]: I agree, that would be the best solution.

File: src/background.ts (hunk 2 above)
Comment (Sneezry) [reply src/background.ts]: If the extension is locked, and the user presses the hotkey to auto-fill code, this code block can prevent the command. Do I understand correctly?

Comment (mymindstorm) [reply src/background.ts]: Yes, but doesn't this also prevent unencrypted access?

Comment (Sneezry) [reply src/background.ts]: Oh, I got your point. Let me update this.

Comment (Sneezry) [reply src/background.ts]: Updated

**Comment by Haiyangha** [comment]
gh pr checkout 831

**Comment by Haiyangha** [comment]
gh pr checkout 831

File: src/background.ts (hunk 1 above)
Comment (Jeunesse16) [reply src/background.ts:595]: Birden fazla eşleşen giriş varsa, kullanıcıdan her zaman doğru olanı seçmesini istememiz gerektiğini düşünüyorum, ancak ilkini değil. Ne düşünüyorsun?

**Comment by Jeunesse16** [comment]
Birden fazla eşleşen giriş varsa, kullanıcıdan her zaman doğru olanı seçmesini istememiz gerektiğini düşünüyorum, ancak ilkini değil. Ne düşünüyorsun?

**Pull request merged**